lib/features/family/features/bedtime/presentation/widgets/bedtime_slider_widget.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:givt_app/core/enums/enums.dart';
+import 'package:givt_app/features/family/features/bedtime/presentation/models/bedtime.dart';
+import 'package:givt_app/features/family/shared/design/components/input/input.dart';
+import 'package:givt_app/features/family/shared/widgets/texts/shared_texts.dart';
+import 'package:givt_app/features/family/utils/family_app_theme.dart';
+import 'package:givt_app/utils/utils.dart';
+
+class BedtimeSliderWidget extends StatefulWidget {
+  BedtimeSliderWidget({
+    this.initialAmount,
+    this.onAmountChanged,
+    super.key,
+  });
+
+  final double? initialAmount;
+  final void Function(double amount)? onAmountChanged;
+  @override
+  State<BedtimeSliderWidget> createState() => _BedtimeSliderWidgetState();
+}
+
+class _BedtimeSliderWidgetState extends State<BedtimeSliderWidget> {
+  late double currentAmount;
+
+  @override
+  void initState() {
+    super.initState();
+    currentAmount = widget.initialAmount ?? BedtimeConfig().defaultBedtimeHour;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        Container(
+          alignment: Alignment.center,
+          child: HeadlineLargeText(
+            '${currentAmount.floor()}:${getMinutes(currentAmount)} pm',
+            color: FamilyAppTheme.primary99,
+          ),
+        ),
+        SliderTheme(
+          data: FunSliderTheme.getSliderTheme(context),
+          child: Slider(
+            min: BedtimeConfig().minBedtimeHour,
+            value: currentAmount,
+            max: BedtimeConfig().maxBedtimeHour,
```

Comment:
I'm not 100% sure what your intention is with the BedTimeConfig. Because it seems like you're using it as a static holder of configuration values but everywhere you are instantiating an object of it. Is it supposed to be a instantiable class?

Reply:
yea you are right i will change it to static